src/components/TransactionsTracker/useTransactionsTracker.ts

```diff
@@ -1,40 +1,40 @@
 import { useEffect, useRef } from 'react';
 import { getTransactionsByHashes as defaultGetTxByHash } from 'apiCalls/transactions';
 import { GetTransactionsByHashesType } from 'types';
-import { useCheckTransactionStatus, useGetPendingTransactions } from 'hooks';
-import { getIsTransactionPending } from 'utils';
+import {
+  useCheckTransactionStatus,
+  useGetAccountInfo,
+  useGetNetworkConfig,
+  useRegisterWebsocketListener
+} from 'hooks/index';
 
 export interface TransactionsTrackerType {
   getTransactionsByHash?: GetTransactionsByHashesType;
 }
 
 export function useTransactionsTracker(props?: TransactionsTrackerType) {
-  const { pendingTransactionsArray } = useGetPendingTransactions();
+  const { network } = useGetNetworkConfig();
+  const { address } = useGetAccountInfo();
+  const checkTransactionStatus = useCheckTransactionStatus();
+  const checkTransactionStatusRef = useRef(checkTransactionStatus);
 
   const getTransactionsByHash =
     props?.getTransactionsByHash ?? defaultGetTxByHash;
 
-  const checkTransactionStatus = useCheckTransactionStatus();
-  const intervalRef = useRef<any>();
-
-  const pendingBatches = pendingTransactionsArray.filter(
-    ([sessionId, transactionBatch]) => {
-      const isPending =
-        sessionId != null && getIsTransactionPending(transactionBatch.status);
-      return isPending;
-    }
-  );
+  const onMessage = () => {
+    checkTransactionStatusRef.current({
+      shouldRefreshBalance: true,
+      getTransactionsByHash
+    });
+  };
 
   useEffect(() => {
-    if (pendingBatches.length > 0) {
-      intervalRef.current = setInterval(() => {
-        checkTransactionStatus({ getTransactionsByHash });
-      }, 2000);
-    } else {
-      clearInterval(intervalRef.current);
-    }
-    return () => {
-      clearInterval(intervalRef.current);
-    };
-  }, [pendingBatches]);
+    checkTransactionStatusRef.current = checkTransactionStatus;
+  }, [checkTransactionStatus]);
+
+  useRegisterWebsocketListener({
```

Comment:
There's a tricky situation here: this is a hook, and if you call this hook multiple times or maybe in different components, this will run multiple times, opening a lot of connections. 

We need to rethink this a bit: either run this hook only once per mounting the app, or, even better, inside the useRegisterWebsocketListener, make a list of listeners with keys to them and don't register a listener if it already has one